go-controller/pkg/ovn/pods.go

```diff
@@ -26,31 +27,44 @@ func (oc *DefaultNetworkController) syncPods(pods []interface{}) error {
 	//
 	// TBD: Before this succeeds, add Pod handler should not continue to allocate IPs for the new Pods.
 	expectedLogicalPorts := make(map[string]bool)
+	vms := make(map[ktypes.NamespacedName]bool)
 	for _, podInterface := range pods {
 		pod, ok := podInterface.(*kapi.Pod)
 		if !ok {
 			return fmt.Errorf("spurious object in syncPods: %v", podInterface)
 		}
 
-		if !oc.isPodScheduledinLocalZone(pod) {
+		switchName, err := oc.getExpectedSwitchName(pod)
+		if err != nil {
+			return fmt.Errorf("failed geting expected switch name at syncPods: %v", err)
+		}
+		expectedLogicalPortName := ""
+		var annotations *util.PodAnnotation
+		if kubevirt.IsPodLiveMigratable(pod) {
+			vms, expectedLogicalPortName, annotations, err = oc.allocateSyncMigratablePodIPsOnZone(vms, pod)
+			if err != nil {
+				return err
+			}
+		} else if oc.isPodScheduledinLocalZone(pod) {
+			expectedLogicalPortName, annotations, err = oc.allocateSyncPodsIPs(switchName, pod)
+			if err != nil {
+				return err
+			}
+		} else {
 			continue
 		}
 
-		annotations, err := util.UnmarshalPodAnnotation(pod.Annotations, ovntypes.DefaultNetworkName)
-		if err != nil {
+		if annotations == nil {
 			continue
 		}
-		expectedLogicalPortName, err := oc.allocatePodIPs(pod, annotations, ovntypes.DefaultNetworkName)
-		if err != nil {
-			return err
-		}
+
 		if expectedLogicalPortName != "" {
 			expectedLogicalPorts[expectedLogicalPortName] = true
 		}
 
 		// delete the outdated hybrid overlay subnet route if it exists
 		newRoutes := []util.PodRoute{}
-		switchName := pod.Spec.NodeName
+		switchName = pod.Spec.NodeName
```

Comment:
There is someting confusing about initializing above `switchName, err := oc.getExpectedSwitchName(pod)` and then overwriting here with `switchName = pod.Spec.NodeName`. You shsouldn't need to do this as they should be the same thing. Since this is the `DefaultNetworkController` you can just use `pod.Spec.NodeName`

Reply:
Make sense, going back to use node.Spec.NodeName where it make sense, also note that at the last commit the `switchName` variable can be removed.